### PR TITLE
#534 | Redesign - Block page

### LINKED ISSUE #534: Block route redesign
the `/block` route should be refactored to show the following information _instead of_ a table of transactions:

- block height
    - tooltip: `Also called "Block Number". The block height represents the length of of the blockchain, increasing by one after the addition of a new block`
- timestamp
    - tooltip: "The time and date the block was produced"
    - in the format of `[clock icon] X unit ago (Mmm-DD-YYYY HH:MM:SS AM/PM +UTC)` e.g. `9 mins ago (Feb-16-2024 10:06:23 PM +UTC)`
- transactions
    -tooltip: "The number of transactions in the block. Internal Transactions are transactions which occur as a result of contract execution and include a TLOS value"
    - in the format of `X transactions and Y contract internal transactions in this block`, where "X transactions" is clickable and leads to `/txs?block={blocknum}`
- size
    - tooltip: "the amount of data in the block, determined by the gas limit"
    - `X bytes` where X is the size of the block, formatted using `toLocaleString()`
- gas used
    - tooltip: "The amount of gas used in the block"
    - show gas used, formatted using `toLocaleString()`
- gas limit
    - tooltip: "The gas limit for all transactions in the block"
    - show gas limit, formatted using `toLocaleString()`
- nonce
    - tooltip: "The value used during mining to achieve consensus on Proof of Work for the block"
    - show nonce
- hash
    - tooltip: "The hash of the block header"
    - show hash
- parent hash
    - tooltip: "The hash of the parent block"
    - show hash

### PR DESCRIPTION
**Pull request by Viterbo**
# Fixes #534 

## Description

This PR implements the new design for the Block page, including two tabs that can be parametrized in the URL.

This PR also includes the [Base redesign PR](https://github.com/telosnetwork/teloscan/pull/538) from @ezra-sg.

## Test scenarios
- Visit [this block page](https://deploy-preview-542--teloscan-redesign.netlify.app/block/283647239?tab=overview) on the overview tab
  - on that view, click on Prev and Next buttons to the side of the height block number
    - the block should change
- Visit another Block but [on the transactions tab](https://deploy-preview-542--teloscan-redesign.netlify.app/block/283647108?tab=transactions)
- Change the language in the menu and check all texts are (at least) present 

## Screenshots
image

image

|image|image|
----------------------------------

## Checklist:

-   [x] I have performed a self-review of my own code
-   [x] I have commented my code, particularly in hard-to-understand areas
-   [x] I have cleaned up the code in the areas my change touches
-   [x] My changes generate no new warnings
-   [x] Any dependent changes have been merged and published in downstream modules
-   [x] I have checked my code and corrected any misspellings
-   [x] I have removed any unnecessary console messages
-   [x] I have included all english text to the translation file and/or created a new issue with the required translations for the currently supported languages
-   [x] I have tested for mobile functionality and responsiveness
-   [x] I have added appropriate test coverage

### CODE REVIEW

File: src/pages/BlockPage.vue [hunk 1]
```diff
@@ -0,0 +1,424 @@
...
+        align="left"
+    >
+        <q-tab class="p-block__tabs-tab" name="overview" :label="$t('pages.blockpage.overview')" />
+        <q-tab class="p-block__tabs-tab" name="transactions" :label="$t('pages.blockpage.transactions')" />
+    </q-tabs>
+
+    <div class="p-block__main-container">
+        <div class="p-block__main-content">
+            <q-tab-panels v-model="tab" class="p-block__panels">
+                <q-tab-panel class="p-block__panel" name="overview">
```
Comment (Viterbo) [inline src/pages/BlockPage.vue:178]: The content of this panel can be cleaned up by creating a new Component for the Block (in fact, two components, one for each card.

What do you think about creating a whole pair of components? is it worth it?

File: src/pages/BlockPage.vue [hunk 2]
```diff
@@ -0,0 +1,424 @@
+<!-- eslint-disable max-len -->
+<!-- eslint-disable max-len -->
+<!-- eslint-disable max-len -->
+<!-- eslint-disable max-len -->
+<!-- eslint-disable @typescript-eslint/no-explicit-any -->
+<!-- eslint-disable no-unused-vars -->
```
Comment (ezra-sg) [inline src/pages/BlockPage.vue:6]: you can remove all of these (i checked locally, no errors)

File: src/pages/BlockPage.vue (hunk 1 above)
Comment (ezra-sg) [reply src/pages/BlockPage.vue:178]: yes, i think creating child components would be a good idea 👍  this one is a bit large as it is

File: src/pages/BlockPage.vue [hunk 3]
```diff
@@ -0,0 +1,424 @@
...
+                } catch (e) {
+                    console.error(e);
+                    return gas.toString();
+                }
+            }
+            return 0;
+        },
+        gasLimit() {
+            if (this.blockData) {
+                console.log('this.blockData.gasLimit', typeof this.blockData.gasLimit, this.blockData.gasLimit);
```
Comment (ezra-sg) [inline src/pages/BlockPage.vue:76]: remove log

**Review by ezra-sg — COMMENTED** [review]
nice work! I think a few changes are needed:
- i think the css class names in BlockPage.vue are off, instead of `p-block` i think they should be `c-block` (sorry 😭) to match other components
- there should be a singular and plural version for the text "X transactions in this block", like where there is 1 it should say "1 transaction in this block"
- can you make the info icons (for the tooltips) and the labels gray instead of black/white? i think especially in light mode they look a little intense
- instead of 2 separate cards for the parts of the page, i think they should be one card with an `<hr>` line separating them
image
- we need to decide how we want to have tabs appear with the redesign. this one is up for discussion because we have no designs for it. I personally think they should be styled to look more like other block explorers, like this but using our colors, what do you think?
image

**Comment by ezra-sg** [comment]
also lets default to the plain english text (like "an hour ago") for timestamp rather than the long form
image

**Comment by donnyquixotic** [comment]
> also lets default to the plain english text (like "an hour ago") for timestamp rather than the long form image

on hover, we can display the full date timestamp:
image

**Review by donnyquixotic — CHANGES_REQUESTED** [review]
@Viterbo please refactor BlockPage.vue to use `setup` attribute (`<script lang='ts' setup>`). See https://github.com/telosnetwork/teloscan/pull/540/files for refactor example and telos wallet for other component examples.

File: src/pages/BlockPage.vue [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+        blockData.value = toRaw(response.data?.results?.[0]) as BlockData;
+    } catch (error) {
+        console.error('Failed to fetch block data:', error);
+        blockData.value = null;
+    }
+};
+
+// Watchers
+
+// cuando block cambie hay que actualizar la url
```
Comment (pmjanus) [inline src/pages/BlockPage.vue:56]: English comments please

**Review by pmjanus — APPROVED** [review]

File: src/pages/BlockPage.vue [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+    router.push({ name: 'block', params: { block: parseInt(block.value) - 1 } });
+}
+
+function nextBlock() {
+    router.push({ name: 'block', params: { block: parseInt(block.value) + 1 } });
+}
+
+function visitNativeBlockExplorer(extraData) {
+    console.log('visitNativeBlockExplorer()', extraData);
+    window.open(`https://explorer-test.telos.net/block/${extraData}`, '_blank');
```
Comment (Viterbo) [inline src/pages/BlockPage.vue:38]: I forgot to put this URL in the env file. I will tomorrow morning

File: src/pages/BlockPage.vue (hunk 4 above)
Comment (donnyquixotic) [inline src/pages/BlockPage.vue:56]:
```suggestion
// update url on block change
```

**Review by donnyquixotic — APPROVED** [review]

File: src/pages/BlockPage.vue (hunk 5 above)
Comment (donnyquixotic) [reply src/pages/BlockPage.vue]: I took care of it

File: src/pages/BlockPage.vue (hunk 4 above)
Comment (donnyquixotic) [reply src/pages/BlockPage.vue]: updated

**Pull request merged**